app/actions/DaemonActions.js

```diff
@@ -1,49 +1,89 @@
 import {versionCheckAction} from "./WalletLoaderActions";
 import * as daemon from "../wallet/daemon";
+import { RPCDaemonHost} from "config";
 
 export const DAEMONSTARTED = "DAEMONSTARTED";
+export const DAEMONSTARTED_ADVANCED = "DAEMONSTARTED_ADVANCED";
+export const DAEMONSTARTED_ADVANCED_ERROR = "DAEMONSTARTED_ADVANCED_ERROR";
 export const DAEMONSTARTED_ERROR = "DAEMONSTARTED_ERROR";
 export const DAEMONSTOPPED = "DAEMONSTOPPED";
 export const DAEMONSTOPPED_ERROR = "DAEMONSTOPPED_ERROR";
 export const DAEMONSYNCING_START = "DAEMONSYNCING_START";
 export const DAEMONSYNCING_PROGRESS = "DAEMONSYNCING_PROGRESS";
 export const DAEMONSYNCED = "DAEMONSYNCED";
 export const WALLETREADY = "WALLETREADY";
+export const LOADER_ADVANCED_SUCCESS = "LOADER_ADVANCED_SUCCESS";
 
-export const startDaemon = (rpcuser, rpcpassword) => (dispatch) => daemon
-  .startDaemon(rpcuser, rpcpassword)
-  .then(pid => {
+export const startDaemon = () => (dispatch) => {
+  daemon.startDaemon()
+  .then(res => {
+    const {pid, advancedDaemon} = res;
     dispatch({type: DAEMONSTARTED, pid});
-    dispatch(syncDaemon(rpcuser, rpcpassword));
+    const next = advancedDaemon ? {type: DAEMONSTARTED_ADVANCED, advancedDaemon} : syncDaemon();
+    dispatch(next);
   })
   .catch(() => dispatch({type: DAEMONSTARTED_ERROR}));
+};
+
+export const startDaemonAdvanced = (args) => (dispatch) => {
+  const {rpcuser, rpcpassword, rpccert } = args;
+  const rpchost = RPCDaemonHost();
+  const credentials = {
+    rpcuser: rpcuser,
+    rpcpassword: rpcpassword,
+    rpccert: rpccert
+  };
+  daemon.startDaemonAdvanced(args)
+  .then( () => {
+    dispatch(syncDaemon(credentials, rpchost));
```

Comment:
I'd prefer you store these credentials in state for use else where instead of continually passing them further and further down.

Reply:
In the end, seems like you are passing these all the way down just to have them accessible for the startRpc request.